### DISCUSSION TITLE
Recommended versions for `@web3modal/ethereum`, `@web3modal/react` & `wagmi`

### DISCUSSION THREAD

**Discussion by 0xhohenheim** [WalletConnect/walletconnect-monorepo#2417] [Developer Support] [answered]
Hey guys!

So I was recently working with web3modal + wagmi in React and faced an issue with the `useChainId` hook where the `chainId` would only return the chains that were configured and wouldn't return the accurate `chainId` of the connected wallet. After looking up the documentation of `wagmi` I realised `useChainId` was most likely replaced by `useNetwork` which worked perfectly.

I then decided to update `wagmi` so I wouldn't use deprecated hooks and `useContract` & `useSigner` weren't available anymore, both of which were critical to my DApp, I couldn't rework that.

My current dependencies that work fine:
```json
{
    "@web3modal/ethereum": "^2.2.1",
    "@web3modal/react": "^2.2.1",
    "ethers": "5.6.9",
    "wagmi": "^0.12.2",
    "web3": "^1.5.3"
}
```

But what are the recommendations? I would very much like to keep using `useContract`, the hooks from the wagmi documentation - `useContractRead` & `useContractWrite` doesn't seem as friendly to me.

**Comment by xzilja** [comment, marked as answer]
Wagmi released major change via v1, which re-worked a lot of hooks and under the hood replaced ethers with  their new library https://viem.sh To continue using latest versions of wagmi, you'll need to go over their migration guide. In terms of web3modal, version 2.4.x and up requires wagmi v1, below that you can still use previous version of wagmi

**Reply by 0xhohenheim** [reply]
Are there any web3modal examples with wagmi v1 that can be used as a reference?

**Reply by xzilja** [reply]
It's essentially same as before, with few changes around how you do initial implementation i.e. as per updated docs https://docs.walletconnect.com/2.0/web3modal/react/installation#implementation

Everything else is just normal wagmi usage, so best check out https://wagmi.sh

**Reply by 0xhohenheim** [reply]
Right, the [migration guide](https://wagmi.sh/react/migration-guide) also clears up a few things. Thanks!
